File: src/main/java/net/gooday2die/MIPSim/Engine/MachineCodes/MachineCode.java
package net.gooday2die.MIPSim.Engine.MachineCodes;

/**
 * A class that represents a single machine code
 * R Type instructions will be returned as 1
 * I Type instructions will be returned as 2
 * J Type instructions will be returned as 3
 * @author Gooday2die (Isu Kim)
 */
public class MachineCode {
    private final int code; // the machine code to set
    private final int codeType;
    /**
     * A constructor method for class MachineCode
     * @param machineCode the machine code integer to set
     * @param argCodeType the type to set
     */
    public MachineCode(int machineCode, int argCodeType){
        code = machineCode;
        codeType = argCodeType;
    }

    /**
     * A getter method for attribute code
     * @return returns integer value of code
     */
    public int getCode() {
        return code;
    }

    /**
     * A getter method for attribute code Type
     * @return returns type object of this current machine code
     */
    public int getCodeType(){
        return codeType;
    }
}

File: MachineCodes/MachineCode.cpp
//
// @file : MachineCode.cpp
// @author : Gooday2die (Isu Kim)
// Contacts : edina00@naver.com
// @brief : A file that implements all MachineCode member functions
// @date: 2022-03-07

#include "MachineCode.h"

MachineCode::MachineCode(int argCode, uint8_t argType) {
    this->code = argCode;
    this->codeType = argType;
}

uint8_t MachineCode::getCodeType() {
    return this->codeType;
}

int MachineCode::getcode() {
    return this->code;
}